### PR TITLE
Update to IDF 5.3.1

### PR DESCRIPTION
**Pull request by tannewt**
And update I2C timeouts to be tick agnostic

### CODE REVIEW

File: ports/espressif/common-hal/busio/I2C.c [hunk 1]
```diff
@@ -73,9 +73,6 @@ void common_hal_busio_i2c_construct(busio_i2c_obj_t *self,
         mp_raise_ValueError(MP_ERROR_TEXT("All I2C peripherals are in use"));
     }
 
-    // Delete any previous driver.
-    i2c_driver_delete(self->i2c_num);
-
```
Comment (dhalbert) [inline ports/espressif/common-hal/busio/I2C.c:78]: Could you say why you removed this? Thanks.

File: ports/espressif/esp-idf-config/sdkconfig-debug.defaults [hunk 2]
```diff
@@ -15,6 +15,8 @@ CONFIG_COMPILER_OPTIMIZATION_SIZE=y
 #
 CONFIG_ESP_SYSTEM_PANIC_PRINT_HALT=y
 CONFIG_ESP_PANIC_HANDLER_IRAM=y
+CONFIG_ESP_CONSOLE_SECONDARY_NONE=y
+# CONFIG_ESP_CONSOLE_SECONDARY_USB_SERIAL_JTAG is not set
```
Comment (dhalbert) [inline ports/espressif/esp-idf-config/sdkconfig-debug.defaults:19]: Is this new in 5.3.1, or does it fix some existing issue?

**Review by dhalbert — CHANGES_REQUESTED** [review]
Just some questions!

File: ports/espressif/esp-idf-config/sdkconfig-debug.defaults (hunk 2 above)
Comment (tannewt) [reply ports/espressif/esp-idf-config/sdkconfig-debug.defaults:19]: I think 5.3 switched the default for USB_SERIAL_JTAG to true. I've been setting this manually for debug builds before.

File: ports/espressif/common-hal/busio/I2C.c (hunk 1 above)
Comment (tannewt) [reply ports/espressif/common-hal/busio/I2C.c:78]: I was seeing extra logging that this was failing. I think init/deinit code does this correctly.

**Review by dhalbert — APPROVED** [review]
Thanks!

**Pull request merged**